apps/web/src/components/providers/EnvironmentProvider.tsx

```diff
@@ -143,25 +135,38 @@ export function EnvironmentProvider({ children }: { children: React.ReactNode })
   );
 
   useEffect(() => {
-    (async () => {
-      if (environments) {
-        await switchEnvironment({ environmentId: getEnvironmentId() });
-      }
-    })();
-  }, [environments, switchEnvironment]);
+    if (environments && environments.length > 0 && !currentEnvironment) {
+      switchEnvironment({ environmentId: getEnvironmentId() });
+    }
+  }, [currentEnvironment, environments, switchEnvironment]);
```

Comment:
This change is to avoid calling switchEnvironment on every path change - now its being called just once when `currentEnvironment` doesn't exist which is only upon registration - in other cases we are switching env directly using switchEnvironment functions.